src/Features/Core/Portable/RemoveUnusedMembers/AbstractRemoveUnusedMembersDiagnosticAnalyzer.cs

```diff
@@ -366,6 +359,52 @@ PooledHashSet<ISymbol> GetCandidateSymbolsReferencedInDocComments(INamedTypeSymb
                 return builder;
             }
 
+            ArrayBuilder<string> GetDebuggerDisplayAttributeArguments(INamedTypeSymbol namedTypeSymbol)
+            {
+                var builder = ArrayBuilder<string>.GetInstance();
+                AddDebuggerDisplayAttributeArguments(namedTypeSymbol, builder);
+                return builder;
+            }
+
+            void AddDebuggerDisplayAttributeArguments(INamedTypeSymbol namedTypeSymbol, ArrayBuilder<string> builder)
+            {
+                AddDebuggerDisplayAttributeArgumentsCore(namedTypeSymbol, builder);
+
+                foreach (var member in namedTypeSymbol.GetMembers())
+                {
+                    switch (member)
+                    {
+                        case INamedTypeSymbol nestedType:
+                            AddDebuggerDisplayAttributeArguments(nestedType, builder);
+                            break;
+
+                        case IPropertySymbol property:
+                        case IFieldSymbol field:
+                            AddDebuggerDisplayAttributeArgumentsCore(member, builder);
+                            break;
+                    }
+                }
+            }
+
+            void AddDebuggerDisplayAttributeArgumentsCore(ISymbol symbol, ArrayBuilder<string> builder)
+            {
+                foreach (var attribute in symbol.GetAttributes())
+                {
+                    if (attribute.AttributeClass == _debuggerDisplayAttributeType &&
+                        attribute.ConstructorArguments.Length == 1 &&
+                        attribute.ConstructorArguments[0] is var arg &&
+                        arg.Kind == TypedConstantKind.Primitive &&
+                        arg.Type.SpecialType == SpecialType.System_String)
+                    {
+                        var value = arg.Value as string;
+                        if (value != null)
```

Comment:
nit: technically could just use pattern matching and life this into the if clause.